diff --git a/modules/skparagraph/src/ParagraphImpl.h b/modules/skparagraph/src/ParagraphImpl.h
--- a/modules/skparagraph/src/ParagraphImpl.h	(revision f204b137b97b44b7397de173fc54181c37ac6501)
+++ b/modules/skparagraph/src/ParagraphImpl.h	(date 1659467114710)
@@ -126,7 +126,7 @@

     size_t lineNumber() override { return fLines.size(); }

-    TextLine& addLine(SkVector offset, SkVector advance,
+    TextLine& addLine(SkVector offset, SkScalar indent, SkVector advance,
                       TextRange textExcludingSpaces, TextRange text, TextRange textIncludingNewlines,
                       ClusterRange clusters, ClusterRange clustersWithGhosts, SkScalar widthWithSpaces,
                       InternalLineMetrics sizes);