diff --git a/modules/skparagraph/src/ParagraphStyle.cpp b/modules/skparagraph/src/ParagraphStyle.cpp
--- a/modules/skparagraph/src/ParagraphStyle.cpp	(revision f204b137b97b44b7397de173fc54181c37ac6501)
+++ b/modules/skparagraph/src/ParagraphStyle.cpp	(date 1659467114706)
@@ -19,6 +19,11 @@
     fEnabled = false;
 }

+TextIndent::TextIndent() {
+    fFirstLine = 0.0;
+    fRestLine = 0.0;
+}
+
 ParagraphStyle::ParagraphStyle() {
     fTextAlign = TextAlign::kStart;
     fTextDirection = TextDirection::kLtr;