diff --git a/modules/skparagraph/src/TextWrapper.h b/modules/skparagraph/src/TextWrapper.h
--- a/modules/skparagraph/src/TextWrapper.h	(revision f204b137b97b44b7397de173fc54181c37ac6501)
+++ b/modules/skparagraph/src/TextWrapper.h	(date 1659467114715)
@@ -176,6 +176,7 @@
                                                   size_t startClip,
                                                   size_t endClip,
                                                   SkVector offset,
+                                                  SkScalar indent,
                                                   SkVector advance,
                                                   InternalLineMetrics metrics,
                                                   bool addEllipsis)>;